### PR TITLE
split data export archive paths on provider uuid

### PR DESCRIPTION
**Pull request by infinitewarp**
for #1315

### CODE REVIEW

File: koku/masu/util/upload.py [hunk 1]
```diff
@@ -13,13 +13,16 @@
...
 
-def get_upload_path(account_name, provider_type, date, table_name, daily=False):
+def get_upload_path(
+    account_name, provider_type, provider_uuid, date, table_name, daily=False
+):
     """
     Get the s3 upload_path for a file.
 
     Args:
         account_name (str): Koku user account (schema) name.
```
Comment (werwty) [inline koku/masu/util/upload.py]: Small suggestion: s/account_name/schema_name/

**Comment by infinitewarp** [comment]
demo of current functionality: https://asciinema.org/a/279862

File: koku/api/dataexport/syncer/__init__.py [hunk 2]
```diff
@@ -94,40 +97,70 @@ def _copy_object(self, s3_destination_bucket, source_object):
...
                 for source_object in self.s3_source_bucket.objects.filter(
-                    Prefix=(
-                        f'{settings.S3_BUCKET_PATH}/{account}/{provider_type}/'
-                        f'{day.year:04d}/{day.month:02d}/{day.day:02d}/'
-                    )
+                    Prefix=prefix
                 ):
                     self._copy_object(s3_destination_bucket, source_object)
+
+            log_uuid = str(uuid.uuid4())
```
Comment (werwty) [inline koku/api/dataexport/syncer/__init__.py]: Is the point of this uuid to keep track of a single sync task?
If so shouldn't we be using the uuid declared above?

File: koku/masu/celery/tasks.py [hunk 3]
```diff
@@ -228,15 +75,29 @@ def upload_normalized_data():
     accounts, _ = Orchestrator.get_accounts()
 
     # Deduplicate schema_name since accounts may have the same schema_name but different providers
```
Comment (werwty) [inline koku/masu/celery/tasks.py]: This comment is no longer valid. Accounts is built from the list of providers: https://github.com/project-koku/koku/blob/master/koku/masu/external/accounts/db/cur_accounts_db.py#L76 so we don't need deduplication. 

You could remove
```
    schema_providers = set(
        (account['schema_name'], account['provider_uuid']) for account in accounts
    )
```
and iterate through the accounts. This save only a marginal amount of time so I completely understand if you want to leave as is.

File: koku/api/dataexport/syncer/__init__.py (hunk 2 above)
Comment (infinitewarp) [reply koku/api/dataexport/syncer/__init__.py]: Good catch! Fixing this in the next push.

File: koku/masu/celery/tasks.py (hunk 3 above)
Comment (infinitewarp) [reply koku/masu/celery/tasks.py]: TIL! I'll prune this out with the next push.

**Pull request merged**